src/routes/docs/tutorials/android/step-4/+page.markdoc

```diff
@@ -0,0 +1,509 @@
+---
+layout: tutorial
+title: Authentication
+description: Learn to build an Android app with no backend code using an Appwrite backend.
+step: 4
+difficulty: beginner
+---
+
+# User Flow {% #user-flow %}
+
+This is the user flow that we are going to achieve in this page
+
+![user-flow](/images/docs/tutorials/recipewrite-user-flow.png)
+
+Keeping the userflow in mind, let us start by initializing three new empty activities -
+
+- **LoginActivity**
+- **RegisterActivity**
+- **HomeActivity**
+
+# Appwrite methods {% #appwrite-methods %}
+
+Let us implement some utility methods in **Appwrite.java**
+
+- **onGetAccount**
+
+It will be used in **MainActivity** to check if the user is already logged in or not.
+
+```java
+public static void onGetAccount(Context context) {
```

Comment:
Missing imports. These should not be in Appwrite.js tbh. Why is there UI logic in Appwrite.java, which is just a service provider for Appwrite client?

Reply:
You are right, I apologize for having a weak understanding here. I have removed all of the UI logic from `Appwrite.java`. However, I still think that having `Appwrite.java` is helpful as it will serve as the one place where all IDs and endpoints related to Appwrite are kept. LMK if you feel otherwise.